api/src/main/java/org/pmiops/workbench/api/ProfileController.java

```diff
@@ -672,37 +672,52 @@ private boolean userNeverLoggedIn(com.google.api.services.admin.directory.model.
   @AuthorityRequired({Authority.ACCESS_CONTROL_ADMIN})
   public ResponseEntity<EmptyResponse> bypassAccessRequirement(Long userId, String moduleName, AccessBypassRequest request) {
     Timestamp valueToSet;
+    Timestamp previousValue;
     Boolean bypassed = request.getIsBypassed();
+    User user = userDao.findUserByUserId(userId);
     if (bypassed) {
       valueToSet = new Timestamp(clock.instant().toEpochMilli());
     } else {
       valueToSet = null;
     }
     switch (moduleName) {
       case "dataUseAgreement":
+        previousValue = user.getDataUseAgreementBypassTime();
         userService.setDataUseAgreementBypassTime(userId, valueToSet);
         break;
       case "complianceTraining":
+        previousValue = user.getComplianceTrainingBypassTime();
         userService.setComplianceTrainingBypassTime(userId, valueToSet);
         break;
       case "betaAccess":
+        previousValue = user.getBetaAccessBypassTime();
         userService.setBetaAccessBypassTime(userId, valueToSet);
         break;
       case "emailVerification":
+        previousValue = user.getEmailVerificationBypassTime();
         userService.setEmailVerificationBypassTime(userId, valueToSet);
         break;
       case "eraCommons":
+        previousValue = user.getEraCommonsBypassTime();
         userService.setEraCommonsBypassTime(userId, valueToSet);
         break;
       case "idVerification":
+        previousValue = user.getIdVerificationBypassTime();
         userService.setIdVerificationBypassTime(userId, valueToSet);
         break;
       case "twoFactorAuth":
+        previousValue = user.getTwoFactorAuthBypassTime();
         userService.setTwoFactorAuthBypassTime(userId, valueToSet);
         break;
       default:
         throw new BadRequestException("There is no access module named: " + moduleName);
     }
+    userService.logAdminUserAction(
+        userId,
+        "updated user bypass on " + moduleName,
```

Comment:
Wording nit: maybe "Set bypass status for module " + moduleName + " to " + bypassed

I think including the true/false value is useful since otherwise an admin (who I imagine might eventually be reading these log messages in a table in the UI somewhere) would have to grok the previousValue / valueToSet which may not be immediately clear how to interpret.